src/mavlink-router/endpoint.cpp

```diff
@@ -537,6 +541,16 @@ int UartEndpoint::open(const char *path)
         goto fail;
     }
 
+    /* For linux set poll latency at chip level */
+#ifdef __linux__
+    {
+        struct serial_struct serial_ctl;
+        ioctl(fd, TIOCGSERIAL, &serial_ctl);
```

Comment:
Can you check the return value?

Reply:
> @catch-twenty-two I'm designating you the official UART master at Auterion for now. Could you please make sure to cross-check if all of the settings that we do here are applied?
> https://github.com/mavlink/c_uart_interface_example/blob/master/serial_port.cpp#L359-L496
> 
> A lot of developers have no idea what they are doing when it comes to UARTs, so every single time one gets used, we need to audit the config. I have seen this way too often in my career and further failure is not acceptable.

Sound Good! Sure, I can take a look.

Reply:
Done

Reply:
upstream pr is here: https://github.com/intel/mavlink-router/pull/165